# Papime pe101019
### Última modificación: 09-Septiembre-2020, 11:42:00
### Autor: Maria Fernanda Ocampo 
# Matrices 

¿Sabias que el uso de las matrices es algo que se puede incorporar a la vida cotidiana?, es algo de gran utilidad y que además puede hacer que el resolver un problema sea mas sencillo.
En el siguiente material se hablará sobre la aplicación y se abordaran algunos temas referentes a matrices.
##  **Problema inicial**
Obtenido del libro Álgebra Lineal octava edición de Bernard Kolman.
#### **Cálculo de la calificación promedio de un curso**
 
 Suponga que un profesor utiliza cuatro notas para determinar la calificación promedio que obtiene un estudiante en un curso: cuestionarios, dos exámenes de una hora y un examen final. Cada una de estas notas tiene una ponderación de 10%, 30%, 30% y 30%, respectivamente. 
 
 Si las calificaciones de un estudiante son, en cada rubro, 78, 84, 62 y 85,
 
  ¿Como podemos calcular el promedio del curso haciendo uso de matrices?



### Introducción
 
En la vida diaria el concepto de matrices es de gran relevancia, ya que las matrices se usan como contenedores para almacenar datos relacionados, en la vida cotidiana se puede utilizar en todo lo relacionado a solución de ecuaciones por ejemplo en economía se utilizan para hacer análisis de producción, distribución y organización de las empresas, en teoría de juegos, biología y hasta en psicología, en ingeniería robotica se pueden usar para cálcular el movimiento de un brazo mecánico, otra aplicación es el reconocimiento facial. 

Una matriz es un conjunto ordenado en una estructura de filas y columnas, los elementos de este conjunto pueden ser objetos matemáticos de diferentes tipos, sin embargo,  trabajaremos exclusivamente con matrices formadas por números reales.

Cada uno de los números  que conforman la matriz se denomina elemento y se denotan por $(a_{ij})$ donde i representa el número de fila y j el número de la columna en que se encuentra el elemento de la matriz A, el número de filas y columnas de una matriz respresenta la  dimensión de una matriz, por ejemplo una matriz A de n filas y m columnas podemos denotarla como $A_{nxm}$, es decir: 

 \begin{equation} A_{nxm} =
\begin{pmatrix}
a_{11} & a_{12} & \cdots & a_{1m}\\
a_{21} & a_{22} & \cdots & a_{2m}\\
\vdots & \vdots & \ddots & \vdots\\
a_{n1} & a_{n2} & \cdots & a_{nm}
\end{pmatrix}
\end{equation}


A continuación se presentará un ejemplo en el que identificaremos la dimensión de la matriz, asi como sus elementos.

* Ejemplo:

In [ ]:
#En este ejemplo se hará una matriz de números aleatorios del (0,20) en donde nos pedirá el número de columnas
#y el número de renglones que queremos.
import random  
import numpy as np
#pedimos el número de columnas y número de filas que queremos que tenga la matriz
m = int(input('Dame el número de columnas m para la matriz: '))
n = int(input('Dame el número de filas n  para la matriz: '))

#creamos nuestra matriz de nxm
A=  np.zeros((n,m))
for i in range(n):
    for j in range(m):
    #damos el intervalo en el que pueden tomar valores cada elemento de la matriz.
        A[i][j] = random.randint(0,20) 

Dame el número de columnas  para la matriz: 3
Dame el número de filas  para la matriz: 2


In [ ]:
print(A)

[[15.  1. 20.]
 [ 0.  3. 19.]]


En la matriz anterior que denotamos A, podemos observar que es una matriz de dimensión nxm, en el siguiente ejemplo definiremos la matriz con valores fijos para analizar mejor.


In [ ]:
#La matriz que daremos será la que se muestra 
import numpy as np
print("Matriz A de dimensión 2x3, :")

A= np.array([[11,10,18],[17,18,1]])

print(A)


Matriz A de dimensión 2x3, :
[[11 10 18]
 [17 18  1]]


Es decir:

 \begin{equation} A_{2x3}=
\begin{pmatrix}
11 & 10 & 18 \\
17 & 18 & 1 
\end{pmatrix}
\end{equation}

In [ ]:
#obtendremos el elemento b11
a11= A[0][0]
print("El elemento a11 es:")
print(a11)

el elemento a11 es:
11


In [ ]:
#obtenemos el elemento b3,4
a22= A[1][1]
print("El elemento a22 es:")
print(a22)

El elemento a22 es:
18


Dicho de otra manera, tenemos que:
* $a_{11}$= 11
* $a_{22}$= 18

## Tipos de matrices 

Exiten tipos de matrices dependiendo el tipo de elementos que la conforme y su orden, cabe mencionar que todas se pueden crear en jupyter con la paqueteria numpy.

Algunas de las matrices que se conocen son las que se mencionan a continuación:

***Por elementos***

* Matriz nula o matriz cero : es aquella cuyos elementos son todos cero.
* Matriz identidad: es aquella que cumple la propiedad de ser el elemento neutro del producto de matrices, es de nxn y tiene como elementos el 1 en la diagonal (los elementos que forman parte de la diagonal son  $a_{11}$, $a_{22}$, $a_{33}$,...,$a_{nn}$) y cero en aquellos elementos que no pertenecen a la diagonal.
* Matriz diagonal:  Una matriz se denomina diagonal si todos los elementos que
no se encuentran sobre la diagonal son cero.
* Matriz triangular superior:  todas sus componentes abajo de la diagonal son cero y los que estan en la diagonal y por encima de esta son casi o todos distintos de cero.
* Matriz real: todos sus elementos pertenecen a los números reales
* Matriz compleja : todos sus elementos pertenecen a los números complejos.


***Por orden:***
* Matriz cuadrada: es aquella que cumple la condición de que n=m, en este caso se denota $A_{n}$
* Matriz rectangular: aquella que cumple la condición de que n $\neq$ m que fue el caso inical.

La igualdad de matrices se presenta cuando para dos matrices A,B se cumple siempre que $a_{ij}$=$b_{ij}$ para todo (i,j).


## Operaciones con matrices


- Suma  y Resta de matrices 

Para llevar a cabo estas operaciones es importante verificar que las matrices con las que se trabajará tengan el mismo orden, es importante saber que se obtendrá como resultado una matriz con exactamente ese orden al realizar las operaciones.

La suma o resta se lleva a cabo elemento por elemento, por ejemplo sean A y B dos matrices que deseamos sumar, entonces se obtendrá una nueva matriz C cuyos elementos serán: $c_{ij}$=$a_{ij}+b_{ij}$, para hacer la resta solo le aplica la operación correspondiente.

De manera gráfica, podemos verlo de la siguiente manera para la suma:

\begin{equation} A_{nxm} = 
\begin{pmatrix}
a_{11} & a_{12} & \cdots & a_{1m}\\
a_{21} & a_{12} & \cdots & a_{2m}\\
\vdots & \vdots & \ddots & \vdots\\
a_{n1} & a_{n2} & \cdots & a_{nm}
\end{pmatrix}
\end{equation}


 \begin{equation} B_{nxm} =
\begin{pmatrix}
b_{11} & b_{12} & \cdots & b_{1m}\\
b_{21} & b_{12} & \cdots & b_{2m}\\
\vdots & \vdots & \ddots & \vdots\\
b_{n1} & b_{n2} & \cdots & b_{nm}
\end{pmatrix}
\end{equation}


 \begin{equation} C=A+B =
\begin{pmatrix}
a_{11}+b_{11} & a_{1x}+b_{12} & \cdots & a_{1m}+b_{1m}\\
a_{21}+b_{21} & a_{12}+b_{12} & \cdots & a_{2m}+ b_{2m}\\
\vdots & \vdots & \ddots & \vdots\\
a_{n1}+b_{n1} & a_{n2}+b_{n2} & \cdots & a_{nm}+b_{nm}
\end{pmatrix}
\end{equation}



In [ ]:
#ejemplo de suma de matrices 
print("Recuerde que las matrices a sumar deben tener el mismo número de filas entre ellas y el mismo número de renglones")
#pedimos que nos de el número de filas y columnas que tienen las matrices
filas = int(input ("Indique el número de filas: "))
columnas = int(input ("Indique el número de columnas: "))
#creamos las matrices en las que se guardarán los valores que el usuario proporcione y los de la suma
matriz1 = np.zeros((filas,columnas))
matriz2 = np.zeros((filas,columnas))
#le pedimos que ingrese la primer matriz elemento por elemento y vamos llenando
print ('Ingrese la primer matriz')
for i in range(filas):
		for j in range(columnas):
			matriz1[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print( "su primer matriz es:")
print(matriz1)
#le pedimos que ingrese la segunda matriz elemento por elemento y vamos llenando           
print ('Ingrese la segunda matriz')
for i in range(filas):
	for j in range(columnas):
			matriz2[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print("su segunda matriz es:")
print( matriz2)
#realizamos la suma de las matrices que nos proporciono y la imprimimos 
matriz3= matriz1+matriz2
print ('La suma de las matrices es:')
print (matriz3)

Recuerde que las matrices a sumar deben tener el mismo número de filas entre ellas y el mismo número de renglones
Indique el número de filas: 2
Indique el número de columnas: 2
Ingrese la primer matriz
Elemento (0,0): 2
Elemento (0,1): 2
Elemento (1,0): 2
Elemento (1,1): 2
su primer matriz es:
[[2. 2.]
 [2. 2.]]
Ingrese la segunda matriz
Elemento (0,0): 2
Elemento (0,1): 2
Elemento (1,0): 2
Elemento (1,1): 2
su segunda matriz es:
[[2. 2.]
 [2. 2.]]
La suma de las matrices es:
[[4. 4.]
 [4. 4.]]


De lo anterior se puede observar como el elemento(0,0) de nuestra nueva matriz es la suma del elemento (0,0) de las dos matrices que introdujimos inicialmente.
Cabe destacar que en esta operación con matrices se mantiene la asociatividad, conmutatividad, el emento neutro que seria la matriz cuyas entradas son todas cero y por último el elemento opuesto que si tenemos la matriz A, el elemento opuesto  sería -A  que es una matriz cuyos elementos son los opuestos a A.

- Escalar por una matriz

En este caso lo que se hace es multiplicar un número escalar por la matriz, es decir $\tau$*$A_{nxm}$ y el resultado será una matriz del mismo orden pero con cada uno de sus elementos múltiplicado por el escalar,es decir: 


  \begin{equation} \tau*A_{nxm}=
\begin{pmatrix}
\tau*a_{11} & \tau*a_{12} & \cdots & \tau*a_{1m}\\
\tau*a_{21} & \tau*a_{12} & \cdots & \tau*a_{2m}\\
\vdots & \vdots & \ddots & \vdots\\
\tau*a_{n1} & \tau*a_{n2} & \cdots & \tau*a_{nm}
\end{pmatrix}
\end{equation}

Cabe mencionar que esta operación es distributiva respecto a la suma de escalares y a la suma de matrices, además es conmutativa y asociativa respecto al producto punto de escalares.
A continuación se muestra un ejemplo de esta operación.


In [ ]:
#pedimos que nos de el número de filas y columnas que tienen las matrices
import numpy as np
filas = int(input ("Indique el número de filas: "))
columnas = int(input ("Indique el número de columnas: "))
#creamos las matrices en las que se guardarán los valores que el usuario proporcione y los de la suma
matriz1 = np.zeros((filas,columnas))
#le pedimos que ingrese la primer matriz elemento por elemento y vamos llenando
print ('Ingrese la  matriz')
for i in range(filas):
        for j in range(columnas):
            matriz1[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print( "su matriz es:")
print(matriz1)
#le pedimos que ingrese el escalar por el que se multiplicara           
l= int(input('Dame el número por el que multiplicarás la matriz'))
#realizamos la multiplicación de la matricez que nos proporciono por el escalar y la imprimimos 
matriz2= matriz1*l
print ('La matriz multiplicada por el escalar es:')
print (matriz2)

Indique el número de filas: 2
Indique el número de columnas: 2
Ingrese la  matriz
Elemento (0,0): 2
Elemento (0,1): 2
Elemento (1,0): 2
Elemento (1,1): 2
su matriz es:
[[2. 2.]
 [2. 2.]]
Dame el número por el que multiplicarás la matriz2
La matriz multiplicada por el escalar es:
[[4. 4.]
 [4. 4.]]


- Porducto de un vector por una matriz 

Si tenemos una matriz $A_{nxm}$ y queremos obtener el producto de esta matriz con un vector, lo primero que se debe verificar es que el número de columnas de la matriz coincida con el número de filas del vector, el resultado de esto nos dara una nueva matriz $C_{nx1}$
Los elementos que conformarán esta nueva matriz serán el resultado de sumar los productos de los elementos de la fila i por el vector, es decir:

Sea la matriz:
  \begin{equation}  A_{nxm} =
\begin{pmatrix}
a_{11} & a_{12} & \cdots & a_{1m}\\
a_{21} & a_{12} & \cdots & a_{2m}\\
\vdots & \vdots & \ddots & \vdots\\
a_{n1} & a_{n2} & \cdots & a_{nm}
\end{pmatrix}
\end{equation}

Definiremos el vector:

  \begin{equation}  V_{mx1} =
\begin{pmatrix}
a_{11} \\
a_{21} \\
\vdots \\
a_{m1} 
\end{pmatrix}
\end{equation}

Entonces el producto de estos será la nueva matriz $C_{nx1}$ cuyas entradas serán:

 

\begin{equation}
  C_{i1} = \sum_{k} (a_{ik}*b_{k,1})
\end{equation}

In [ ]:
import numpy as np
#primero pedimos que se nos den la matriz
filas1 = int(input ("Indique el número de filas de la matriz: "))
columnas1 = int(input ("Indique el número de columnas de la matriz: "))
columnas2 = 1

#creamos la matriz en la que se guardarán los valores que el usuario proporcione
A = np.zeros((filas1,columnas1))    
B = np.zeros((columnas1,columnas2))   
 #llenamos la matriz con los datos del usuario
print ("Ingrese la primer matriz")

for i in range(filas1):
        for j in range(columnas1):
            A[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print( "su matriz es:")
print(A)
#llenamos el vector
print ("Ingrese el vector")
for i in range(columnas1):
        for j in range(columnas2):
            B[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print( "su vector es:")
print(B)


C = A.dot(B)
print("El producto de la matriz con el vector es:")
    
print (C)


Indique el número de filas de la matriz: 2
Indique el número de columnas de la matriz: 2
Ingrese la primer matriz
Elemento (0,0): 2
Elemento (0,1): 2
Elemento (1,0): 2
Elemento (1,1): 2
su matriz es:
[[2. 2.]
 [2. 2.]]
Ingrese el vector
Elemento (0,0): 2
Elemento (1,0): 2
su vector es:
[[2.]
 [2.]]
El producto de la matriz con el vector es:
[[8.]
 [8.]]


- Producto de Matrices

Si tenemos una matriz $A_{nxm}$ y queremos obtener el producto de esta matriz con una $B_{mxl}$, lo primero que se debe verificar es que el número de columnas de la matriz A coincida con el número de filas de la matriz B, el resultado de esto nos dara una nueva matriz $C_{nxl}$.

Los elementos que conformarán esta nueva matriz serán el resultado de sumar los productos de los elementos de la fila i por la columna j, es decir:

\begin{equation}
  C_{ij} = \sum_{k} (a_{ik}*b_{k,j})
\end{equation}

por ejemplo si tenemos las siguientes matrices:


\begin{equation} A_{2x3}=
\begin{pmatrix}
2 & 3 & 1 \\
0 & -1 & -2 
\end{pmatrix}
\end{equation}


\begin{equation} B_{3x2}=
\begin{pmatrix}
 1 & 0 \\
-1 & 2  \\
-2 & 3
\end{pmatrix}
\end{equation}


\begin{equation} C_{2x2}=
\begin{pmatrix}
 (2*1)+(-1*3)+(-2*1)   & (0*2)+(2*3) +(3*1) \\
 (1*0)+(-1*-1)+(-2*-2) & (0*0)+(2*-1)+(3*-2)
\end{pmatrix}
\end{equation}



\begin{equation} =
\begin{pmatrix}
 -3 & 9 \\
  5 & -8
 \end{pmatrix}
 \end{equation}
 
 
 Esta operación cumple con la asociatividad, es distributiva respecto a la suma de matrices, no siempre es conmutativo, y no se cumple necesariamente que si A*B=A*C  no necesariamente se cumple que B=C.

In [ ]:
import numpy as np
#primero pedimos que se nos den las matrices 
print("recuerde que el numero de columnas de la segunda matriz debe ser igual al número de columnas de la primer matriz")
filas1 = int(input ("Indique el número de filas de la primer matriz: "))
columnas1 = int(input ("Indique el número de columnas de la primer matriz: "))
columnas2 = int(input ("Indique el número de columnas de la segunda matriz: "))

#creamos las matrices en las que se guardarán los valores que el usuario proporcione y los de la suma
A = np.zeros((filas1,columnas1))    
B = np.zeros((columnas1,columnas2))   
 #llenamos la matriz 1 con los datos del usuario
print ("Ingrese la primer matriz")

for i in range(filas1):
        for j in range(columnas1):
            A[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print( "su matriz 1 es:")
print(A)
#llenamos la matriz 2
print ("Ingrese la segunda matriz")
for i in range(columnas1):
        for j in range(columnas2):
            B[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print( "su matriz 2 es:")
print(B)


C = A.dot(B)
print("El producto de las matrices es:")
    
print (C)


recuerde que el numero de columnas de la segunda matriz debe ser igual al número de columnas de la primer matriz
Indique el número de filas de la primer matriz: 2
Indique el número de columnas de la segunda matriz: 2
Indique el número de columnas de la segunda matriz: 2
Ingrese la primer matriz
Elemento (0,0): 2
Elemento (0,1): 2
Elemento (1,0): 2
Elemento (1,1): 2
su matriz 1 es:
[[2. 2.]
 [2. 2.]]
Ingrese la segunda matriz
Elemento (0,0): 3
Elemento (0,1): 3
Elemento (1,0): 3
Elemento (1,1): 3
su matriz 2 es:
[[3. 3.]
 [3. 3.]]
El producto de las matrices es:
[[12. 12.]
 [12. 12.]]


- Matriz Inversa 
Una matriz tiene cuadrada $A_{nxn}$ tiene inversa denotada como $A^{-1}$ si se cuemple que:

$$A*A^{-1}=A^{-1}A=I$$

De donde I es la matriz identidad de tamaño n, esta matriz tiene unos en su diagonal y ceros en todas las demas entradas.
Las matrices que tienen inversa son llamadas matrices invertibles o regulares.

La matriz inversa debe cumplir:
 * Es única 
 * $(A^{-1})^{-1}$ = A
 * $(A*B)^{-1}$ = $B^{-1}$*$A^{-1}$
 * ($\alpha*A^{-1}$)=$\frac{1}{\alpha}$*$A^{-1}$   con $\alpha$ distinto de 0

In [ ]:
import numpy as np
filas1 = int(input ("Indique el número de filas y columnas de la matriz: "))

A = np.zeros((filas1,filas1))      
 #llenamos la matriz 1 con los datos del usuario
print ("Ingrese la primer matriz")

for i in range(filas1):
        for j in range(filas1):
            A[i][j] = float(input('Elemento (%d,%d): ' % (i, j)))
print( "su matriz es:")
print(A)
#Calculamos la matriz inversa
B= np.linalg.inv(A)

print(" La matriz Inversa es: ")
print(B)

Indique el número de filas y columnas de la matriz: 2
Ingrese la primer matriz
Elemento (0,0): 3
Elemento (0,1): 3
Elemento (1,0): 4
Elemento (1,1): 5
su matriz es:
[[3. 3.]
 [4. 5.]]
 La matriz Inversa es: 
[[ 1.66666667 -1.        ]
 [-1.33333333  1.        ]]



- Aplicaciones 

 -obtenidas del libro Álgebra Lineal octava edición de Bernard Kolman.


Un fabricante de cierto producto realiza tres modelos, A, B y C. Algunas
partes  de cada uno se elaboran en la fábrica $F_1$, ubicada en de Taiwán, y después se terminan en la fábrica $F_2$ de Estados Unidos. El costo total de cada producto consta
de los costos de manufactura y de embarque. En consecuencia, los costos (en dólares) de
cada fábrica pueden describirse mediante las matrices $F_1$ y $F_2$ de 3 × 2:

\begin{equation}
F_1=
\begin{pmatrix}
 manufactura &  embarque  \\  
\hline 
32& 40 & mod.A \\
50& 80 & mod.B \\
70& 20 & mod.C \\
\end{pmatrix}
\end{equation}


Para el caso de Estados Unidos se tiene:


\begin{equation}
F_2=
\begin{pmatrix}
 manufactura &  embarque  \\  
\hline 
40& 60 & mod.A \\
50& 50 & mod.B \\
130& 20 & mod.C \\
\end{pmatrix}
\end{equation}


  El costo total de un producto dependiendo el modelo estaría dado por $F_1$ + $F_2$, que es: 

  \begin{equation}
F_1+F_2=
\begin{pmatrix}
 manufactura &  embarque  \\  
\hline 
80& 120 & mod.A \\
100& 130 & mod.B \\
200& 40 & mod.C \\
\end{pmatrix}
\end{equation}

De lo anterior se observa que para el modelo B, los costos totales son 100  por manufactura y 130 por embarque.






**Aplicación a ecología **

Una siembra se rocía con pesticidas para eliminar insectos dañinos; sin embargo, las plantas absorben parte de las sustancias. Luego, los animales herbívoros de
la zona comen las plantas contaminadas y absorben los pesticidas. Para determinar la
cantidad de pesticida absorbida por uno de esos animales, procedemos de la manera siguiente. Suponga que tenemos tres pesticidas y cuatro plantas. Sea $a_{ij}$ la cantidad de
pesticida i (en miligramos) absorbida por la planta j. Esta información puede representarse mediante la matriz


\begin{equation}
A=
\begin{pmatrix}
planta 1 & planta 2 & planta 3 & planta 4 \\  
\hline 
2& 3& 4& 3& pesticida 1 \\
3& 2& 2& 5& pesticida 2 \\
4& 1& 6& 4& pesticida 3 \\
\end{pmatrix}
\end{equation}


Imagine ahora, que tenemos tres animales herbívoros, y sea $b_{ij}$ la cantidad de plantas del tipo i que uno de ellos, de tipo j, come mensualmente. La información puede representarse mediante la matriz.

\begin{equation}
B=
\begin{pmatrix}
Herbívoro 1 & Herbívoro 2 & Herbívoro 3  \\  
\hline 
20& 12& 8& planta 1 \\
28& 15& 15& planta 2 \\
30& 12& 10& planta 3 \\
40& 16& 20& planta 4
\end{pmatrix}
\end{equation}


Entonces al obtener la matriz A*B, que es la siguiente: 






In [ ]:
#La matriz que daremos será la que se muestra 
import numpy as np
print("Matriz A*B:")

A= np.array([[2,3,4,3],[3,2,2,5],[4,1,6,4]])
B= np.array([[20,12,8],[28,15,15],[30,12,10], [40,16,20]])
C=A.dot(B)
print(C)

Matriz A*B:
[[364 165 161]
 [376 170 174]
 [448 199 187]]


De donde se obtiene por ejemplo que se requieren 174 mg de pesticida 2 absorbidos por el herbívoro 3, que es la entrada (2,3) de la matriz A*B 
Ahora bien, si tuviéramos p animales carnívoros (como el hombre) que se comen a los
herbívoros, podríamos repetir el análisis para determinar cuánto pesticida absorbe cada
uno. ■

##**Solución al problema inicial**

Suponga que un profesor utiliza cuatro notas para determinar la calificación promedio que obtiene un estudiante en un curso: cuestionarios, dos exámenes de una hora y un examen final. Cada una de estas notas tiene una ponderación de 10%, 30%, 30% y 30%, respectivamente. 
 
 Si las calificaciones de un estudiante son, en cada rubro, 78, 84, 62 y 85,
 
  ¿Como podemos calcular el promedio del curso haciendo uso de matrices?

  Lo anterior se resuelve con una multiplicacion de matrices, ya que si en una matriz colocamos las ponderaciones y en otra la calificación correspondiente sera facíl obtener el promedio, es decir:

Sea P la matriz de ponderaciones y C la de calificaciones, entonces estas se verán de la siguente manera:


\begin{equation} P =
\begin{pmatrix}
.10  \\
.30   \\
.30    \\
.30    \\
\end{pmatrix}
\end{equation}

\begin{equation} C =
\begin{pmatrix}
78 & 84 & 62 & 85
\end{pmatrix}
\end{equation}

De donde se obtendrá que C*P= (78*.10)+ (84*.30)+(62*.30)+(85*.30)= 7.8+ 25.2+ 18.6 + 25.5 = 77.1

Entoces de lo anterior podemos ver que el resultado es 77.1 de promedio.

Fuentes que se consultaron y quedan como sugerencia al lector: 

+ Kolman, B., & Hill, D. R. (2006). Álgebra Lineal. México: Pearson.

+ S., S. I., & Godoy, J. J. (2012). Álgebra Lineal. México: McGrawll -Hill Companies.

